src/main.c

```diff
@@ -11,7 +11,7 @@ int train(int batch_size, int ndim, int *dims_hidden, int epochs, double learnin
     dims[0] = dataset.rows * dataset.cols;
     for (int i = 1; i < ndim - 1; i++)
     {
-        dims[i] = dims_hidden[i];
+        dims[i] = dims_hidden[i - 1];
```

Comment:
then you also have to adjust:

https://github.com/tiny-technologies/neural/blob/8b6ab9af98db4b3e7934b09d8adede1717fcca29/src/main.c#L125-L127